### ISSUE TITLE
[Bug]: game doesn't launch on Mac

### ISSUE THREAD

**Issue by primfandango** [MelvorIdle/melvoridle.github.io#4034]
### Describe the bug

When I try to launch the game via Steam on Mac, a new icon comes up for it and Steam says "running", but I can't click on the Melvor icon to view the loading screen.

### Reproduction Steps

Open Steam
Click on Library 
Click the play button for Melvor Idle
Game says running but only the icon launches

### Expected behaviour

For the game to open

### Screenshots

<img width="368" alt="Screenshot 2024-01-25 at 11 03 04" src="https://github.com/MelvorIdle/melvoridle.github.io/assets/157694080/de961aad-b0d4-4e33-b48e-e3c6ef904f80">
<img width="214" alt="Screenshot 2024-01-25 at 11 03 12" src="https://github.com/MelvorIdle/melvoridle.github.io/assets/157694080/362b058a-1a5f-4811-8504-e4b0ec97f950">

### Which platforms are you experiencing this bug on?

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari
- [X] Steam
- [ ] Epic Games
- [ ] Mobile App (iOS)
- [ ] Mobile App (Android)
- [ ] Mobile App (Android Beta)
- [ ] Other (Please Specify)

### Which expansions are purchased and active at the time of the bug?

- [X] Throne of the Herald
- [X] Atlas of Discovery

### Which version of the game are you experiencing this bug on?

v1.2.2

### Game Subversion

I can't see this

### Are you using any mods?

No

**Comment by primfandango** [comment]
Neither of the sets of steps on #mac-client-issues has resolved this issue

**Comment by MelvorIdle** [comment]
Can you try removing the folder located here:

`~/Library/Application Support/Caches/Melvor Idle` (This is different to the one in the discord channel).

Let me know if that solves the issue or not.

**Comment by primfandango** [comment]
Sadly didn't work, even after I uninstalled and reinstalled the game again

**Comment by EvilAdd** [comment]
Same here, did those steps MelvorIdle suggest and the game still don't start in my MacOS Monterey

**Comment by Psevdych** [comment]
@MelvorIdle maybe this one will help somehow:
```
zsh 1001 % ~/Library/Application\ Support/Steam/steamapps/common/Melvor\ Idle/Melvor\ Idle.app/Contents/MacOS/nwjs
[11994:259:0126/125410.497674:ERROR:chrome_browser_cloud_management_controller.cc(162)] Cloud management controller initialization aborted as CBCM is not enabled.
[0126/125410.610486:FATAL:v8_initializer.cc(538)] Error loading V8 startup snapshot file
0   nwjs Framework                      0x00000001256f3152 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 38777906
1   nwjs Framework                      0x00000001256e65c3 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 38725795
2   nwjs Framework                      0x0000000125648604 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 38078692
3   nwjs Framework                      0x00000001272ba8d5 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 67906485
4   nwjs Framework                      0x00000001272ba6f7 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 67906007
5   nwjs Framework                      0x0000000124ac27f5 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 25995989
6   nwjs Framework                      0x0000000124ac151a v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 25991162
7   nwjs Framework                      0x0000000124ac19d3 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 25992371
8   nwjs Framework                      0x0000000120f3d930 ChromeMain + 560
9   nwjs Helper                         0x0000000109ac19b9 main + 377
10  dyld                                0x000000010a33c52e start + 462

...

[0126/125410.999805:WARNING:crash_report_exception_handler.cc(235)] UniversalExceptionRaise: (os/kern) failure (5)
[11994:259:0126/125411.413018:ERROR:network_service_instance_impl.cc(594)] Network service crashed, restarting service.
```
```
zsh 1003 % sw_vers
ProductName:	macOS
ProductVersion:	12.7.2
BuildVersion:	21G1974
```

**Comment by 11uhafnk** [comment]
I have same problem after update yesterday
and many another people in "Steam" discussion https://steamcommunity.com/app/1267910/discussions/0/4140564660379581847/ 

I used mac 
Macos 14.2.1

Steam Beta Branch:  Stable Client
Steam Version:  1705108172

App version: i don't sure where i can check it
file info show 1.1 
<img width="1146" alt="image" src="https://github.com/MelvorIdle/melvoridle.github.io/assets/5215020/f1a7b01f-a1cb-4e19-9582-acf43d802b3f">
<img width="691" alt="image" src="https://github.com/MelvorIdle/melvoridle.github.io/assets/5215020/1c7ab24b-9df4-4d73-8b01-95b50a29b854">


problem: after running application have status "running" but haven't any window
<img width="1259" alt="image" src="https://github.com/MelvorIdle/melvoridle.github.io/assets/5215020/502804e9-64bb-4ec7-846e-84e022f3c500">

I haven't use any mods
installed and active both dlc

I try: 
steam verify files - it show ok
reinstall application - same result
rm userdata files - same result

**Comment by MelvorIdle** [comment]
I've released a potential fix to a Steam Beta branch for Melvor Idle. Please let me know if the update on this branch fixes the launch issue.

It can be access from right clicking Melvor Idle in Steam -> Properties -> Betas -> Beta Participation ->There should be one called `macos-client-fix` (Restart steam if it isn't there).

Once it updates, it will provide two launch options:
1) Launch Melvor Idle
2) Launch Melvor Idle (Intel Macs)

If you run an intel mac, please use the Intel option.

If it doesn't load first try, or produces an error, please uninstall it and then reinstall it and try again (Let me know if you had to do this)

**Comment by primfandango** [comment]
Yes! That worked for me. Had to uninstall and reinstall though

**Comment by steinbdj** [comment]
That worked for me as well.  Didn't have to uninstall first (but had previously done that in an attempt to resolve).

**Comment by yxrind** [comment]
use the Intel option, that worked for me.

**Comment by primfandango** [comment]
Should we keep this version downloaded for now? Are you adding the fix to the regular version of the game?

**Comment by MelvorIdle** [comment]
> Should we keep this version downloaded for now? Are you adding the fix to the regular version of the game?

Keep using this version until I release it to the public within the next 24 hours (I’ll put out an announcement on all social channels + steam). Then you can switch back to the public branch.

**Comment by Nlinzer0113** [comment]
On my M1 I tried the MacOS fix and i am still unable to launch. Mine broke yesterday randomly after closing the game and trying to reopen it.

**Comment by gulinghua** [comment]
> On my M1 I tried the MacOS fix and i am still unable to launch. Mine broke yesterday randomly after closing the game and trying to reopen it.

same on my M1, still can't launch game with neither apple silicon version nor intel version

**Comment by MelvorIdle** [comment]
Have you tried uninstalling it and then reinstalling the game via Steam?

**Comment by gulinghua** [comment]
> Have you tried uninstalling it and then reinstalling the game via Steam?

tried, not working with or without macos-fix

**Comment by MelvorIdle** [comment]
@Nlinzer0113 @gulinghua I've pushed a new update to the `macos-client-fix` steam beta branch. Can you check if the game loads on there? Thanks for your help.

**Comment by MelvorIdle** [comment]
Updated client has been pushed the the live game steam branch.

**Comment by erro-3** [comment]
I was told to post here from Reddit --
I am having issues again on Intel Mac via Steam. It seems whatever new update happened on January 31 broke this fix for me.

So last week (January 24th), I was unable to play on my Intel Mac via Steam, just like everyone else here. Then when the Beta macos-client-fix was released, it worked (able to play again via Steam January 27-30).

**On January 31st it was broken again.** 
My understanding is there was a new update after the January 26 fix. That update seems to have reverted it back to the previous problem.

**What happens?**
Launch Melvor via Steam with Intel option (non-Beta). Game shows as open, but the game window never actually appears, even after several minutes (tested 6+ min waiting).

I have tried uninstalling/reinstalling. I have also tried setting the properties to the previous beta fix and that does not work anymore either.

**Comment by MelvorIdle** [comment]
Geez, MacOS development is a nightmare. One thing works for some, breaks for others. This is unreal.

Can you try the below steps and see if it allows you to load?

> Can you try removing the folder located here:
> 
> `~/Library/Application Support/Caches/Melvor Idle` (This is different to the one in the discord channel).
> 
> Let me know if that solves the issue or not.

**Comment by Psevdych** [comment]
For me on Intel MacOS main and beta versions both works fine. Just update fixed issue.

**Comment by erro-3** [comment]
@MelvorIdle Okay, I tried deleting the cache files, still no luck. Also tried uninstall/reinstall again after and no luck.

Does this have to do with the offline updates y'all are slowly rolling out in stages?

I know Mac isn't always the easiest to maintain, but thank you for the help.

**Comment by MelvorIdle** [comment]
Yeah it does relate to that. We had to update NW.js for each platform (Which Melvor runs on), but MacOS doesn't seem to like it at all. We specifically didn't touch MacOS client until we had to because we were worried something like this would happen.

You can try deleting the Client data folder which will reset the client to a fresh new install. This will sign you out of the cloud and will delete your local saves:

- Navigate to ~/Library/Application Support/Melvor Idle
- Delete everything in this folder.

**Comment by erro-3** [comment]
Uninstalled and deleted `~/Library/Application Support/Melvor Idle` and deleted `~/Library/Application Support/Caches/Melvor Idle` all together. Then installed and still not loading the window.

Not sure if this info is helpful, but in Activity Monitor, I can see `Melvor Idle` as well as `nwjs Helper (GPU)` and 2 of `nwjs Helper` running when I launch the game.

!!! Important Update:
I borrowed a different Mac in the house that is running MacOS Ventura (Intel) and the game DID launch to the login screen.
I am using MacOS Big Sur (Intel) so whatever is going on must specifically have to do with using an older system.

Perhaps there is some sort of incompatibility between the NW.js update and older macs.

**Comment by MelvorIdle** [comment]
Thanks for the information @erro-3 

I'll see if I can get my hands on a Mac that runs Big Sur and will see if I can reproduce the issue + issue a fix.

**Comment by Yao-sss** [comment]
I also run games on steam on MAC OS and these only have one icon, but I can't see any screen
<img width="295" alt="image" src="https://github.com/MelvorIdle/melvoridle.github.io/assets/44462120/c71fa0b7-7c23-4610-b82f-4d04fb939ec0">
<img width="121" alt="image" src="https://github.com/MelvorIdle/melvoridle.github.io/assets/44462120/1565ea57-4a0b-4500-9f4f-e4f6324830fa">

**Comment by galexkeene** [comment]
Still having issues here. Intel Mac, 11.7.10 OS X, game fully updated as of this post (1.3 something). I've never been able to launch properly once. I've tried with or without the "mac fix" Beta, both launch options (get an error for the non-Intel one). The app will launch/be running, but no window or anything ever happens. I have like 6+ "hours" in the game because it's "launching" and I forgot to quit one time and so Steam thinks I've played that much. Did other stuff like multiple re-installs, deleting the contents of the Application Support folder, etc. Never successfully launched once.
